src/connector/src/source/reader/reader.rs

```diff
@@ -181,28 +181,32 @@ impl SourceReader {
 
 #[try_stream(boxed, ok = FsPageItem, error = crate::error::ConnectorError)]
 async fn build_opendal_fs_list_stream<Src: OpendalSource>(lister: OpendalEnumerator<Src>) {
-    let matcher = lister.get_matcher();
-    let mut object_metadata_iter = lister.list().await?;
-
-    while let Some(list_res) = object_metadata_iter.next().await {
-        match list_res {
-            Ok(res) => {
-                if matcher
-                    .as_ref()
-                    .map(|m| m.matches(&res.name))
-                    .unwrap_or(true)
-                {
-                    yield res
-                } else {
-                    // Currrntly due to the lack of prefix list, we just skip the unmatched files.
-                    continue;
+    loop {
+        let matcher = lister.get_matcher();
+        let mut object_metadata_iter = lister.list().await?;
+
+        while let Some(list_res) = object_metadata_iter.next().await {
+            match list_res {
+                Ok(res) => {
+                    if matcher
+                        .as_ref()
+                        .map(|m| m.matches(&res.name))
+                        .unwrap_or(true)
+                    {
+                        yield res
+                    } else {
+                        // Currrntly due to the lack of prefix list, we just skip the unmatched files.
+                        continue;
+                    }
+                }
+                Err(err) => {
+                    tracing::error!(error = %err.as_report(), "list object fail");
+                    return Err(err);
                 }
-            }
-            Err(err) => {
-                tracing::error!(error = %err.as_report(), "list object fail");
-                return Err(err);
             }
         }
+
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
What's this for? Please add a comment

1 second sounds trivial, but considering our barrier interval is just 1 second, we need to be cautious.

Reply:
to avoid unnecessary requests to s3 service. If using s3 source, users don't expect new files to be discovered within seconds. 
The sleep here does not block the barrier because it occurs in the file list stream, before merging with the barrier stream.

Reply:
Shall we use `interval.tick()` instead of `sleep` to handle the incremental files? Thus we can make the update interval longer, like 1min(IIRC for origin s3, it's 10 mins).
Can can we make the update interval configured by user?

Reply:
I see. Agree with @wcy-fdu it can be longer and configurable.

Reply:
https://github.com/risingwavelabs/risingwave/pull/18184
The time interval is now configurable and will be carried into main in this pr